WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsServiceStarter.java

```diff
@@ -0,0 +1,93 @@
+package org.wordpress.android.ui.stats.service;
+
+import android.annotation.TargetApi;
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.PersistableBundle;
+
+import org.wordpress.android.util.AppLog;
+
+public class StatsServiceStarter {
+    public static final String ARG_START_ID = "start-id";
+    private static final int JOB_STATS_SERVICE_ID = 8000;
+    private static int jobId = 0;
+
+    public static void startService(Context context, Bundle originalExtras) {
+        if (context == null || originalExtras == null) {
+            return;
+        }
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            Intent intent = new Intent(context, StatsService.class);
+            intent.putExtras(originalExtras);
+            context.startService(intent);
+        } else {
+            // schedule the JobService here for API >= 26. The JobScheduler is available since API 21, but
+            // it's preferable to use it only since enforcement in API 26 to not break any old behavior
+            ComponentName componentName = new ComponentName(context, StatsJobService.class);
+
+            PersistableBundle extras = passBundleExtrasToPersistableBundle(originalExtras);
+            extras.putInt(ARG_START_ID, getNewStartId());
+
+            JobInfo jobInfo = new JobInfo.Builder(JOB_STATS_SERVICE_ID + jobId, componentName)
+                    .setRequiresCharging(false)
+                    .setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY)
+                    .setOverrideDeadline(0) // if possible, try to run right away
+                    .setExtras(extras)
+                    .build();
+
+            JobScheduler jobScheduler = (JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
+            int resultCode = jobScheduler.schedule(jobInfo);
+            if (resultCode == JobScheduler.RESULT_SUCCESS) {
+                AppLog.i(AppLog.T.STATS, "stats job service > job scheduled");
+            } else {
+                AppLog.e(AppLog.T.STATS, "stats job service > job could not be scheduled");
+            }
+        }
+    }
+
+    @TargetApi(21)
+    private static PersistableBundle passBundleExtrasToPersistableBundle(Bundle originalExtras) {
+        PersistableBundle bundle = new PersistableBundle();
+        if (originalExtras != null) {
```

Comment:
Can't we just iterate on the keys of `originalExtras` and add it to the new `PersistableBundle`?
Maybe we can whitelist the `keys` is we don't want all of them copied over.

Reply:
As discussed on DM, we couldn't find an easy way to copy to `PersistableBundle`, leaving as is